src/project/module.ts

```diff
@@ -115,6 +118,67 @@ export async function loadModule(
 		);
 	}
 
+
+	// Read api routes
+	const apiPath = resolve(modulePath, "api");
+	const expectedApiRoutes = new Set(
+		await glob.glob("*.ts", { cwd: resolve(modulePath, "api") }),
+	);
+
+	const routes = new Map<string, ApiRoute>();
+	if (config.api) {
+		for (const routeName in config.api) {
+			validateIdentifier(routeName, Casing.Snake);
+	
+			// Load script
+			const apiRoutePath = resolve(
+				apiPath,
+				routeName + ".ts",
+			);
+			if (!await exists(apiRoutePath)) {
+				throw new UserError(
+					`API route not found at ${relative(Deno.cwd(), apiRoutePath)}.`,
+					{
+						suggest: "Check the api routes in the module.yaml are configured correctly.",
+						path: moduleConfigPath(modulePath),
+					},
+				);
+			}
+
+			// Get full API route path (including module prefix)
+			const routeConfig = config.api[routeName];
+			const mountPoint = projectModuleConfig.api?.pathPrefix ?? `/api/${name}`;
+			const subpath = "path" in routeConfig ? routeConfig.path : routeConfig.pathPrefix;
+			const routePath = resolve(mountPoint, subpath.replace(/^\//, ""));
```

Comment:
looks like we're normalizing/sanitizing the path here. instead, we should be enforcing the path format to be consistent and throw errors if it's not what we expect. always fail by default.